core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -437,7 +438,9 @@ class ListAccumulator[T] extends AccumulatorV2[T, java.util.List[T]] {
       s"Cannot merge ${this.getClass.getName} with ${other.getClass.getName}")
   }
 
-  override def value: java.util.List[T] = java.util.Collections.unmodifiableList(_list)
+  override def value: java.util.List[T] = _list.synchronized {
+    java.util.Collections.unmodifiableList(new ArrayList[T](_list))
```

Comment:
One last thought ... now that this is cloned, does it need to be in an unmodifiable wrapper? maybe it's still a good idea so that the caller doesn't somehow think modifying the list modifies the accumulator